hypothesis-python/src/hypothesis/internal/conjecture/shrinking/floats.py

```diff
@@ -60,11 +52,12 @@ def short_circuit(self):
         if not math.isfinite(self.current):
             return True
 
-        # If its too large to represent as an integer, bail out here. It's
-        # better to try shrinking it in the main representation.
-        return self.current >= MAX_PRECISE_INTEGER
-
     def run_step(self):
+        # above MAX_PRECISE_INTEGER, all floats are integers. Shrink like one.
+        if self.current > MAX_PRECISE_INTEGER:
+            self.delegate(Integer, convert_to=int, convert_from=float)
+            return
```

Comment:
ah yup, good point. `Integer.shrink` does try things like bit shifts so it won't get completely stuck if `n - 1` fails. It's unclear whether this is worse than master (who knows what the block-level shrinking was doing to the float representation), and since the tests are passing I'm inclined to leave it for later. Something to improve for sure!